src/server/repos/controller.ts

```diff
@@ -49,16 +49,35 @@ export const createMirrorHandler = async ({
       org: publicOrg,
     })
 
-    const exists = await contributionOctokit.rest.repos.get({
-      owner: orgData.data.login,
-      repo: input.newRepoName,
-    })
-
-    if (exists.status === 200) {
-      throw new Error(
-        `Repo ${orgData.data.login}/${input.newRepoName} already exists`,
-      )
-    }
+    await contributionOctokit.rest.repos
+      .get({
+        owner: orgData.data.login,
+        repo: input.newRepoName,
+      })
+      .then((res) => {
+        // if we get a response, then we know the repo exists so we throw an error
+        if (res.status === 200) {
+          reposApiLogger.info(
+            `Repo ${orgData.data.login}/${input.newRepoName} already exists`,
+          )
+
+          throw new Error(
+            `Repo ${orgData.data.login}/${input.newRepoName} already exists`,
+          )
+        }
+      })
+      .catch((error) => {
+        // catch and rethrow the error if the repo already exists
+        if ((error as Error).message.includes('already exists')) {
+          throw error
+        }
+
+        // if there is a real error, then we log it and throw it
+        if (!(error as Error).message.includes('Not Found')) {
```

Comment:
We don’t have to do this check since we throw above, so we can just throw here instead

Reply:
:doh: I’m wrong here and this logic is right, carry on!